src/Library/KeyValueParser.hpp

```diff
@@ -0,0 +1,60 @@
+//
+// Copyright (C) 2015 Red Hat, Inc.
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+// Authors: Daniel Kopecek <dkopecek@redhat.com>
+// Authors: Marek Tamaskovic <mtamasko@redhat.com>
+//
+#pragma once
+#ifdef HAVE_BUILD_CONFIG_H
+  #include <build-config.h>
+#endif
+
+#include <iostream>
+#include <vector>
+#include <string>
+#include <fstream>
+#include <map>
+#include <memory>
+
+namespace usbguard
+{
+
+  struct parsed_t {
+    const std::string lvalue;
+    const std::string rvalue;
+    parsed_t(const std::string a, const std::string b): lvalue(a), rvalue(b) {};
+  };
+
+  class KeyValueParser
+  {
+    std::vector<std::string> keys;
+    std::string separator {""};
+    std::map<std::string, std::string> output_map;
+    bool checkKeyValidity(const std::string& key);
+    virtual bool checkMapValidity();
```

Comment:
What is this method supposed to do? Checking the whole map after it is parsed doesn't seem to be a good solution as the information about the source of an error is lost at that point (information like line or column number).